Update BDP Cloud docs (#315)
snowplow/documentation: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md

---
title: "🆕 Managing data structures with the Data Structures Builder"

> date: "2023-03-01"

Reviewer: Not needed for new pages

Reviewer 2: 👍

sidebar_position: 85
---

:::info
The Data Structures Builder is only available for BDP Cloud Customers
:::

> To create a new Data Structure, first navigate to **Data Structures** in the menu and click the **'Create a Data Structure'** button.

Reviewer: This feels like it’s missing some link to other sections explaining what data structures are for (just imagining someone landing here with no context).

Reviewer 2: this page sits in the Introduction to Tracking Design section, where they can find this into... theoretically they go through each section in order

![](images/data-structures-1.png)

> Populate the general information, such as *Name,* and a *Description* and *Vendor*. *Vendor* allows you to organise your data structures, for example, by teams. Snowplow will automatically generate the Tracking Url to be referenced in your tracking code.

Reviewer: This does not seem to offer much guidance beyond what the UI already expects the user to do. Can we give some examples here, using a realistic data structure someone might want to add?

Reviewer 2: We're going to add a video tutorial soon where we go through an actual example.

> You can then add one or multiple properties. For each property, you can set a name, description, its type, a possible enumeration of allowed values. You can also set additional constraints to define if this property should be optional or mandatory, and if *null* values are allowed.

Reviewer: Examples? Considerations for why something should be mandatory or optional, and when should `null`s be allowed?

Reviewer 2: These are all good points but they shouldn't really sit in the DS Builder page. They are more Tracking Design principles and guidance.

![](images/data-structures-2.png)

> Click save on the Property modal and this will save your property changes.

Reviewer: ```suggestion
Click “Save” on the Property dialog box and this will save your property changes.
```
Or any other replacement of “modal” (which is a very cryptic term for user-facing documentation)

> Clicking Save on the Data Structure page will publish your Data Structure to your Iglu Server.

Reviewer: ```suggestion
Clicking “Save” on the Data Structure page will publish your Data Structure to your Iglu Server.
```

> For information about editing and versioning your Data Structure please see the [Versioning in BDP Cloud section](/docs/understanding-tracking-design//versioning-your-data-structures/cloud/index.md).

Reviewer: ```suggestion
For information about editing and versioning your Data Structure, please see [Versioning](/docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md) in BDP Cloud section.
```

Reviewer 2: 👍

---

:::info

> The Data Structures Builder is only available for BDP Cloud customers, for BDP Enterprise users, see [Managing data structures in the console](/docs/understanding-tracking-design/managing-data-structures/index.md), for Open Source users, see [Managing data structures using Iglu](/docs/pipeline-components-and-applications/iglu/igluctl-2/index.md).

Reviewer: ```suggestion
The Data Structures Builder is only available for BDP Cloud customers.

For BDP Enterprise customers, see [Managing data structures in the console](/docs/understanding-tracking-design/managing-data-structures/index.md). For Open Source users, see [Managing data structures using Iglu](/docs/pipeline-components-and-applications/iglu/igluctl-2/index.md).
```
